service/controller/v3/resource/deployment/deployment.go

```diff
@@ -1,108 +1,41 @@
 package deployment
 
 import (
+	"context"
+
 	azureresource "github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2018-02-01/resources"
 	"github.com/Azure/go-autorest/autorest/to"
 	providerv1alpha1 "github.com/giantswarm/apiextensions/pkg/apis/provider/v1alpha1"
-	"github.com/giantswarm/microerror"
 
 	"github.com/giantswarm/azure-operator/service/controller/v3/key"
 )
 
-const (
-	templateContentVersion = "1.0.0.0"
-)
-
-func getDeploymentNames() []string {
-	return []string{
-		mainDeploymentName,
-	}
-}
-
-func (r Resource) newDeployment(obj providerv1alpha1.AzureConfig, overwrites map[string]interface{}) (azureresource.Deployment, error) {
```

Comment:
Lots of cleanup here because the template magic is separated. Way better now IMO.